src/wallet/wallet2_api.h

```diff
@@ -667,6 +667,15 @@ struct WalletManager
     virtual bool walletExists(const std::string &path) = 0;
 
     /*!
+     * @brief verifyWalletPassword - check if the given filename is the wallet
+     * @param keys_file_name - location of keys file
+     * @param password - password to verify
+     * @param watch_only - verify only view keys?
+     * @return
+     */
+    virtual bool verifyWalletPassword(const std::string &keys_file_name, const std::string &password, const bool watch_only) = 0;
```

Comment:
That might want to be const ? Also, the doc string above seems cut off at "return"